extensions/integration-tests/src/test/schemaCompare.test.ts

```diff
@@ -270,6 +274,230 @@ suite('Schema compare integration test suite @DacFx@', () => {
+			await utils.tryDeleteDB(server, targetDB, ownerUri);
+		}
+	});
+
+	test('Schema compare dacpac to database comparison with publishing all changes and then compare again', async function () {
+		this.timeout(5 * 60 * 1000);
+		const server = await getStandaloneServer();
+		const ownerUri = await getConnectionUri(server);
+		const now = new Date().getTime().toString();
+		const operationId = 'testOperationId_' + now;
+		const targetDB: string = 'ads_schemaCompare_targetDB_' + now;
+
+		try {
+			assert(dacfxService, 'DacFx Service Provider is not available');
+			let result = await dacfxService.deployDacpac(dacpac1, targetDB, true, ownerUri, azdata.TaskExecutionMode.execute);
+
+			assert(result.success === true, 'Deploy database 2 (target) should succeed');
+
+			const source: mssql.SchemaCompareEndpointInfo = {
+				endpointType: mssql.SchemaCompareEndpointType.Dacpac,
+				packageFilePath: dacpac2,
+				serverDisplayName: '',
+				serverName: '',
+				databaseName: '',
+				ownerUri: ownerUri,
+				connectionDetails: undefined
+			};
+			const target: mssql.SchemaCompareEndpointInfo = {
+				endpointType: mssql.SchemaCompareEndpointType.Database,
+				packageFilePath: '',
+				serverDisplayName: '',
+				serverName: server.serverName,
+				databaseName: targetDB,
+				ownerUri: ownerUri,
+				connectionDetails: undefined
+			};
+
+			assert(schemaCompareService, 'Schema Compare Service Provider is not available');
+
+			const schemaCompareResult = await schemaCompareService.schemaCompare(operationId, source, target, azdata.TaskExecutionMode.execute, null);
+			assertSchemaCompareResult(schemaCompareResult, operationId, 4);
+
+			//publish all the changes
+			const publishChangesResult = await schemaCompareService.schemaComparePublishChanges(schemaCompareResult.operationId, server.serverName, targetDB, azdata.TaskExecutionMode.execute);
+			assert(publishChangesResult.success === true, 'Publish changes should complete successfully. But it failed.');
+
+			//verify table Table3 is added
+			const dbConnectionId = await utils.connectToServer({
+				serverName: server.serverName,
+				database: targetDB,
+				userName: server.userName,
+				password: server.password,
+				authenticationTypeName: server.authenticationTypeName,
+				providerName: server.providerName
+			});
+			const dbConnectionOwnerUri = await azdata.connection.getUriForConnection(dbConnectionId);
+			await utils.assertTableCreationResult('dbo', 'Table3', dbConnectionOwnerUri, retryCount);
+
+			//verify no differences exist now
+			const schemaCompareResult2 = await schemaCompareService.schemaCompare(operationId, source, target, azdata.TaskExecutionMode.execute, null);
+			assertSchemaCompareResult(schemaCompareResult2, operationId, 0, true);
+		}
+		finally {
+			await utils.tryDeleteDB(server, targetDB, ownerUri);
+		}
+	});
+
+	test('Schema compare dacpac to database comparison with non-default deployment options', async function () {
+		this.timeout(5 * 60 * 1000);
```

Comment:
I think we need higher timeout than this. in DacFx pipeline I had to set the time out in the pipline to higher to make the test run

Reply:
Increased it to 10mins.